src/app/components/core/filter/add-filter-wizard/add-filter-wizard.component.ts

```diff
@@ -76,43 +65,26 @@ export class AddFilterWizardComponent implements OnInit {
     this.getAvailablePlugins('Filter');
   }
 
-  fetchPluginRequestStarted() {
-    this.ngProgress.start();
-    const requestInProgressEle: HTMLElement = document.getElementById('requestInProgress') as HTMLElement;
-    requestInProgressEle.innerHTML = 'fetching available plugins ...';
-  }
-
-  fetchPluginRequestDone() {
-    this.ngProgress.done();
-
-    if (this.pluginData.length) {
-      const ddnEle: HTMLElement = document.getElementsByClassName('ngx-dropdown-button')[0] as HTMLElement;
-      if (ddnEle !== undefined) {
-        ddnEle.click();
-      }
-    }
-
-    const requestInProgressEle: HTMLElement = document.getElementById('requestInProgress') as HTMLElement;
-    if (requestInProgressEle !== null) {
-      requestInProgressEle.innerHTML = '';
+  public dropdownState() {
+    if (this.isOpen === false) {
+      this.isOpen = true;
+    } else {
+      this.isOpen = false;
```

Comment:
Better to rename toggleDropdownState and make `this.isOpen = ! this.isOpen ` ?  Or `this.isOpen = this.isOpen === false ? true : false`